### DISCUSSION TITLE
Create collection and fields from code

### DISCUSSION THREAD

**Discussion by plcosta** [directus/directus#10620] [Archive]
Hi there! 

I know thats possible to use the [schema migration](https://github.com/directus/directus/pull/7939), but I would like to add new collections using only the code and skipping Directus UI. 

Is there a way to create collections and fields using something like the migrations? 

Cheers!

**Comment by rijkvanzanten** [comment]
Sure thing! You can use custom migrations like you mentioned, or rely on the /collections, /fields, and /relations endpoints to manage the schema 👍🏻

**Reply by plcosta** [reply]
thank you, @rijkvanzanten. Is there any detailed documentation about that? I couldn't find it.

**Reply by rijkvanzanten** [reply]
Yeah, see

https://docs.directus.io/extensions/migrations/

and

https://docs.directus.io/reference/system/collections/
https://docs.directus.io/reference/system/fields/
https://docs.directus.io/reference/system/relations/

**Comment by infomiho** [comment]
@rijkvanzanten What would you suggest if we wanted to add `user_updated` system field via migrations file?

<img width="646" alt="Screenshot 2022-01-31 at 13 48 43" src="https://user-images.githubusercontent.com/2223680/151796397-f5df0568-da84-4c44-a651-e4bd57f9c06e.png">

Doing the following doesn't behave the same way as when doing via UI:
```js
table.uuid("user_updated");
table
    .foreign("user_updated")
    .references("directus_users.id")
    .onUpdate("NO ACTION")
    .onDelete("SET NULL");
```

**Reply by rijkvanzanten** [reply]
You're missing the metadata row from directus_fields. Take a look at what the system normally creates there, and add that as well through `knex('directus_fields').insert({...})` 👍🏻
